src/main/java/dev/slimevr/vr/processor/skeleton/LegTweakBuffer.java

```diff
@@ -258,6 +270,17 @@ public Vector3f getRightFootVelocity(Vector3f vec) {
 		return vec.set(rightFootVelocity);
 	}
 
+	public Vector3f getCenterOfMass(Vector3f vec) {
+		if (vec == null)
+			vec = new Vector3f();
+
+		return vec.set(centerOfMass);
+	}
+
+	public void setCenterOfMass(Vector3f centerOfMass) {
+		this.centerOfMass = centerOfMass.clone();
```

Comment:
Why not just keep a unique vector for caching on `centerOfMass` instead of cloning?

Reply:
Should I change all the other setters as well?

Reply:
That would probably be good, but make sure you change the initialization from
```java
private Vector3f centerOfMass = placeHolderVec;
```
to
```java
private Vector3f centerOfMass = new Vector3f();
```
for all of the ones you change, otherwise you'll be modifying the placeholder value, which is very not good.

Reply:
I changed it, idk why I did it the other way to begin with.